src/Microsoft.Framework.Runtime.Abstractions/Infrastructure/HashCodeCombiner.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Runtime.Infrastructure
+{
+    internal class HashCodeCombiner
```

Comment:
It's similar to aspnet/Razor#449, though in that case the class is public for some reason. If we had a shared project, Razor would still need to solve the problem that caused the type to be made public.